bob-backend/src/core.rs

```diff
@@ -356,4 +359,45 @@ impl Backend {
     pub async fn close_unneeded_active_blobs(&self, soft: usize, hard: usize) {
         self.inner.close_unneeded_active_blobs(soft, hard).await
     }
+
+    pub async fn delete(&self, key: BobKey, with_aliens: bool) -> Result<u64, Error> {
+        let (vdisk_id, disk_path) = self.mapper.get_operation(key);
+        let mut total_count = 0;
+        if let Some(path) = disk_path {
+            trace!("DELETE[{}] try delete normal", key);
+            total_count += self
+                .delete_single(key, Operation::new_local(vdisk_id, path.clone()))
+                .await
+                .map_err(|e| {
+                    debug!("DELETE[{}] delete alien error: {}", key, e);
```

Comment:
Copypaste :) 
BTW I think you should create collection of operations (normal + aliens) and single execution function, so you can map-reduce it and return total count

Reply:
Fixed